pkg/ttnpb/messages.go

```diff
@@ -51,6 +51,20 @@ func (v *ApplicationDownlink_ClassBC) FieldIsZero(p string) bool {
 	panic(fmt.Sprintf("unknown path '%s'", p))
 }
 
+// FieldIsZero returns whether path p is zero.
+func (v *ApplicationDownlink_ConfirmedRetry) FieldIsZero(p string) bool {
+	if v == nil {
+		return true
+	}
+	switch p {
+	case "attempt":
+		return v.Attempt == 0
+	case "max_attempts":
```

Comment:
This should be `v.MaxAttempts == nil`.

`FieldIsZero` is related to the zero value concept of Go, where the zero value of a pointer is `nil`, not the zero value of the value of the pointer.  Using `GetValue` conflates `&{Value: 0}` with `nil`.